### ISSUE TITLE
incompatible with https://github.com/getsentry/raven-ruby

### ISSUE THREAD

**Issue by jlgeering** [davetron5000/gli#250]
```
#!/usr/bin/env ruby
require 'gli'
require 'raven'

include GLI::App

desc 'hello'
command :hello do |c|
    c.action do |global_options,options,args|
        puts "hello"
    end
end

exit run(ARGV)
```

fails with "NoMethodError: private method `desc' called for main:Object"

**Comment by jlgeering** [comment]
see also https://github.com/getsentry/raven-ruby/issues/531

**Comment by jlgeering** [comment]
I don't understand enough ruby magic to know if gli or raven is the culprit

**Comment by davetron5000** [comment]
Looks like the fix proposed in getsentry/raven-ruby#531 worked.

**Comment by jlgeering** [comment]
yes indeed, and this was the fix:

```ruby
require "gli"
require "raven"

class MyApp
  extend GLI::App

  # all the GLI stuff you'd normally put at top-level

end

exit MyApp.run(ARGV)
```
